Sources/MapboxDirections/DirectionsResult.swift

```diff
@@ -29,7 +29,10 @@ open class DirectionsResult: Codable {
         distance = try container.decode(CLLocationDistance.self, forKey: .distance)
         expectedTravelTime = try container.decode(TimeInterval.self, forKey: .expectedTravelTime)
         
-        _directionsOptions = decoder.userInfo[.options] as! DirectionsOptions
+        guard let directionsOptions = decoder.userInfo[.options] as? DirectionsOptions else {
+            throw DirectionsCodingError.missingOptions
```

Comment:
This PR continues to assume that options are passed in as a precondition for decoding, but it throws errors instead of force-unwrapping dictionary entries. Ideally, #382 would incorporate these changes to fix #228.